src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java

```diff
@@ -0,0 +1,75 @@
+
+@Service
+public class ItemOrderedProcessorService {
+
+    private static final String POST_MISSING_IMAGE_CHD_ORDER_URI = "/chd-order-api/missing-image-deliveries";
+
+    private final CHDOrderService chdOrderService;
+
+    public ItemOrderedProcessorService(final CHDOrderService chdOrderService) {
+        this.chdOrderService = chdOrderService;
+    }
+
+    public MissingImageDeliveryRequestApi mapChdItemOrderedToMissingImageDeliveryRequestApi(ChdItemOrdered chdItemOrdered) {
+        MissingImageDeliveryRequestApi missingImageDeliveryRequestApi = new MissingImageDeliveryRequestApi();
+        Item item = chdItemOrdered.getItem();
+        Map<String, String> itemOptions = item.getItemOptions();
+
+        missingImageDeliveryRequestApi.setId(item.getId());
+        missingImageDeliveryRequestApi.setCompanyName(item.getCompanyName());
+        missingImageDeliveryRequestApi.setCompanyNumber(item.getCompanyNumber());
+        missingImageDeliveryRequestApi.setOrderedAt(LocalDateTime.parse(chdItemOrdered.getOrderedAt()));
+        missingImageDeliveryRequestApi.setPaymentReference(chdItemOrdered.getPaymentReference());
+        missingImageDeliveryRequestApi.setFilingHistoryCategory(itemOptions.get("filingHistoryCategory"));
+        missingImageDeliveryRequestApi.setFilingHistoryDate(itemOptions.get("filingHistoryDate"));
+        missingImageDeliveryRequestApi.setFilingHistoryDescription(itemOptions.get("filingHistoryDescription"));
+        missingImageDeliveryRequestApi.setFilingHistoryType(itemOptions.get("filingHistoryType"));
+        missingImageDeliveryRequestApi.setItemCost(item.getTotalItemCost());
+        return missingImageDeliveryRequestApi;
+    }
+
+    public void processItemOrdered(ChdItemOrdered chdItemOrdered) {
+        MissingImageDeliveryRequestApi missingImageDeliveryRequestApi = new MissingImageDeliveryRequestApi();
+        try {
+            missingImageDeliveryRequestApi = mapChdItemOrderedToMissingImageDeliveryRequestApi(chdItemOrdered);
+
+            ApiResponse<MissingImageDeliveryRequestApi> missingImageDeliveryRequestApiResponse =
+                    chdOrderService.createCHDOrder(POST_MISSING_IMAGE_CHD_ORDER_URI, missingImageDeliveryRequestApi);
+
+            if (missingImageDeliveryRequestApiResponse.getStatusCode() != CREATED.value()) {
+                processException(missingImageDeliveryRequestApiResponse.getStatusCode(), missingImageDeliveryRequestApiResponse.toString());
+            }
+        } catch (ApiErrorResponseException ex) {
+            String id = missingImageDeliveryRequestApi.getId() != null ? missingImageDeliveryRequestApi.getId() : "Unknown";
+            String errorResponse = "API Response Error for : "
+                    + id + ", Error response: " + ex.toString();
+            processException(ex.getStatusCode(), errorResponse);
+        }
+    }
+
+    public void processException(int statusCode, String errorResponse) {
```

Comment:
Perhaps `processError()`? Does it need to be public?

Reply:
Yes, I'll change that now, It's not actually processing an exception, it throws it.